### PR TITLE
Add Velocity to replace Speed and Acceleration + Add Drag field and drag.rs example

### PR DESCRIPTION
**Pull request by Rayoule**
Here is the velocity + drag update. I hope everything is in order !

### CODE REVIEW

**Comment by Rayoule** [comment]
Someone knows what happened in the code-style check ?

**Comment by Rayoule** [comment]
Someone knows what happened with the code-style check ?

**Comment by abnormalbrain** [comment]
It just wants some of the code reformatted. You can just run `cargo fmt --all` and commit the changes it makes.

File: src/systems.rs [hunk 1]
```diff
@@ -321,25 +321,43 @@ pub(crate) fn particle_transform(
...
-                    particle.acceleration.at_lifetime_pct(lifetime_pct) * time.raw_delta_seconds();
-                transform.translation += direction.0 * speed.0 * time.raw_delta_seconds();
+
+            let delta_time = match particle.use_scaled_time {
+                true => time.delta_seconds(),
+                false => time.raw_delta_seconds(),
+            };
+
+            // The velocity_direction variable is needed due to its relation with the acceleration being a f32 and not a Vec3
+            let velocity_direction = velocity.0.normalize();
```
Comment (abnormalbrain) [inline src/systems.rs:340]: Vector normalization is quite expensive, so having to do this for every particle for every frame would potentially perform pretty poorly at higher particle counts. Maybe we should just go all the way also replace `acceleration` with a vector as well so we can avoid the normalize call here.

File: src/systems.rs [hunk 2]
```diff
@@ -321,25 +321,43 @@ pub(crate) fn particle_transform(
...
+            velocity.0 +=
+                velocity_direction
+                * particle.acceleration.at_lifetime_pct(lifetime_pct)
+                * delta_time;
+            
+            // Apply drag
+            let current_drag = particle.drag.at_lifetime_pct(lifetime_pct);
+            if current_drag > 0.0 {
+                let drag_force =
+                    ( velocity.0.length() * velocity.0.length() )
```
Comment (abnormalbrain) [inline src/systems.rs:352]: We should use `.length_squared()` here. Computing `.length()` requires doing a very expensive square root as the final step. So `.length_squared()` isn't  `length * length`, it's the length formula without the square root step, which makes it substantially faster.

**Review by abnormalbrain — APPROVED** [review]

**Pull request merged**